### PR TITLE
Add missing Token deprecations

### PR DESCRIPTION
**Pull request by julienfalque**
Ref: https://github.com/FriendsOfPHP/PHP-CS-Fixer/issues/2923#issuecomment-316428641.

### CODE REVIEW

**Comment by keradus** [comment]
please add `@deprecated` annotation where it's missing (clearChanged, isChanged, setContent)

**Comment by julienfalque** [comment]
Done.

Now I'm wondering. Introducing deprecations requires a minor version bump, but this one should have been introduced in 2.4 with other deprecations in the same class. Should I change the target?

**Comment by SpacePossum** [comment]
>requires a minor version bump

Strictly it doesn't, although this might be how SF implements semver.
Master target now will mean this will in 2.4.x I guess, so all good to me :)

**Comment by julienfalque** [comment]
Bumping minor version when introducing a deprecation *is* a [Semver requirement](http://semver.org/#spec-item-7):
> Minor version Y (x.Y.z | x > 0) MUST be incremented if new, backwards compatible functionality is introduced to the public API. It MUST be incremented if any public API functionality is marked as deprecated.

**Comment by SpacePossum** [comment]
sorry about that :/

but what why? so if you forgot a deprecation you're not allowed to patch it? semver is weird sometimes

**Comment by julienfalque** [comment]
That's why I am wondering: should we follow semver strictly and target 2.5 or merge it as a fix for a 2.4 oversight? I would go for 2.4.

File: src/Tokenizer/Token.php [hunk 1]
```diff
@@ -339,9 +341,13 @@ public function isCast()
      * Check if token was changed.
      *
      * @return bool
+     *
+     * @deprecated since 2.5
```
Comment (keradus) [inline src/Tokenizer/Token.php]: 2.4

**Comment by keradus** [comment]
2.4 bugfix

File: src/Tokenizer/Token.php (hunk 1 above)
Comment (julienfalque) [inline src/Tokenizer/Token.php]: Done.

**Comment by SpacePossum** [comment]
we don't want to trigger warning on all those methods?

**Comment by keradus** [comment]
is it missing somewhere ?

**Comment by SpacePossum** [comment]
GH UI is playing mind tricks :+1:

**Comment by SpacePossum** [comment]
please rebase @julienfalque

**Comment by keradus** [comment]
@fabpot, we are experiencing weird behaviour of fabbot.io service.

In `.php_cs` file, we are asking for using `'array_syntax' => array('syntax' => 'short'),` ruleset, yet we get complains that we shall use long form. Could you take a look please?

**Comment by julienfalque** [comment]
Debug tests reverted. Relevant config file that is not correctly taken into account by fabbot.io can be viewed here: https://github.com/julienfalque/PHP-CS-Fixer/blob/e0f3841c5663652e6a371da3acf2adc68b75a42c/.php_cs

**Comment by keradus** [comment]
or directly in PR history commit:
https://github.com/FriendsOfPHP/PHP-CS-Fixer/pull/2924/commits/e0f3841c5663652e6a371da3acf2adc68b75a42c

together with fabbot report:
http://fabbot.io/report/FriendsOfPHP/PHP-CS-Fixer/2924/e0f3841c5663652e6a371da3acf2adc68b75a42c

**Comment by keradus** [comment]
Thank you @julienfalque.

**Pull request closed without merging**